Add example revolving around Twitch data to documentation (#48)
Quantco/datajudge: docs/source/examples/example_twitch.rst

In other words, a 'version' of the data set represents to a temporal notion.
For example, version 1 might stem from end of March and version 2 from end of April.
Moreover, we will assume that the first, version 1, has been vetted and approved with the
help of manual investigation and domain knowledge. The second data set, version 2, has just been
made available. We would like to use it but can't be sure of its validity just yet. As a consequence
we would like to assess the quality of the data in version 2.

The original data set can be found on `kaggle <https://www.kaggle.com/datasets/aayushmishra1512/twitchdata>`_.
For the sake of this tutorial, we slightly process it and provide two version of it.
One can either recreate this by executing this `processing <https://github.com/Quantco/datajudge/tree/main/docs/source/examples/twitch_upload.py>`_
oneself on the original data or download our processed files (`version 1 <https://github.com/Quantco/datajudge/tree/main/docs/source/examples/twitch_version1.csv>`_
and `version 2 <https://github.com/Quantco/datajudge/tree/main/docs/source/examples/twitch_version2.csv>`_) right away.

> The aforementioned script also populates two tables in a Postgres database. This resembles the following:

Reviewer: ```suggestion
The script above also populates two tables in a Postgres database. This resembles the following:
```

     - followers
     - followers_gained
     - views_gained
     - partnered
     - mature
     - language
   * - xQcOW
     - 6196161750
     - 215250
     - 222720
     - 27716
     - 3246298
     - 1734810
     - 93036735
     - True
     - False
     - English
   * - summit1g
     - 6091677300
     - 211845
     - 310998
     - 25610
     - 5310163
     - 1374810
     - 89705964
     - True
     - False
     - English
   * - Gaules
     - 5644590915
     - 515280
     - 387315
     - 10976
     - 1767635
     - 1023779
     - 102611607
     - True
     - True
     - Portuguese
   * - ESL_CSGO
     - 3970318140
     - 517740
     - 300575
     - 7714
     - 3944850
     - 703986
     - 106546942
     - True
     - False
     - English

Note that we expect both version 1 and version 2 to follow this structure. Due to them
being assembled at different points in time, merely their rows shows differ.

Now let's write an actual specification, expressing our expectations against the data.
First, we need to make sure a connection to database can be established at test execution
time. How this is done exactly depends on how you set up your database. When using our

> default setup with running `$ ./start_postgres.sh <https://github.com/Quantco/datajudge/blob/main/start_postgres.sh>`_,

Reviewer: Probably here would be good to mention it's a container-based setup.

Author: Good point! I restructured this a bit to talk about the dockerized setup at the very beginning.

this would look as follows:

.. code-block:: python

    import os
    import pytest
    import sqlalchemy as sa

    @pytest.fixture(scope="module")
    def datajudge_engine():
        address = os.environ.get("DB_ADDR", "localhost")
        connection_string = f"postgresql://datajudge:datajudge@{address}:5432/datajudge"
        return sa.create_engine(connection_string)

> Once a way to connect to database is defined, we want to declare our data sources and

Reviewer: ```suggestion
Once a way to connect to the database is defined, we want to declare our data sources and
```

> express expectations against them. In this example we have two tables in the same database -

Reviewer: ```suggestion
express expectations against them. In this example, we have two tables in the same database -
```

one table per version of the Twitch data.

> Yet, let's start with a very simple example only using version 2. We want to use our

Reviewer: ```suggestion
Yet, let's start with a straightforward example only using version 2. We want to use our
```

> domain knowledge that values of the ``language`` column should only contain letters

Reviewer: ```suggestion
domain knowledge that constrains the values of the ``language`` column only to contain letters
```

   test_result = constraint.test(datajudge_engine)
   >       assert test_result.outcome, test_result.failure_message
   E       AssertionError: tempdb.public.twitch_v2's column(s) 'average_viewers' has mean 4970.2188235294117647, deviating more than 0.1 from tempdb.public.twitch_v2's column(s) 'average_viewers''s  3567.6584158415841584. Condition on second table: WHERE mature IS TRUE;

   /usr/local/Caskroom/miniconda/base/envs/datajudge/lib/python3.10/site-packages/datajudge/pytest_integration.py:25: AssertionError
   ===================================== short test summary info ======================================
   FAILED twitch_specification.py::test_func[VarCharRegex::tempdb.public.twitch_v2] - AssertionError...
   FAILED twitch_specification.py::test_func[KolmogorovSmirnov2Sample::public.twitch_v1 | public.twitch_v2]
   FAILED twitch_specification.py::test_func[UniquesEquality::public.twitch_v1 | public.twitch_v2]
   FAILED twitch_specification.py::test_func[NumericMean::public.twitch_v2 | public.twitch_v2] - Ass...
   =================================== 4 failed, 4 passed in 1.80s ====================================

> So we see that we might not want to trust version 2 of the data as is. What exactly do we

Reviewer: ```suggestion
So we see that we might not want to blindly trust version 2 of the data as is, and that a new exploration might be in order. What exactly do we
```

Author: I was unfamiliar with this usage of the expression 'in order' and took the liberty to rephrase slightly.
